src/nbl/video/CVKLogicalDevice.h

```diff
@@ -124,51 +327,472 @@ class CVKLogicalDevice final : public ILogicalDevice
+
+            const VkMemoryPropertyFlags desiredMemoryProperties =
+                VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT | VK_MEMORY_PROPERTY_HOST_COHERENT_BIT;
+
+            for (uint32_t i = 0u; i < vk_physicalDeviceMemoryProperties.memoryTypeCount; ++i)
+            {
+                const bool isMemoryTypeSupportedForResource = (supportedMemoryTypes & (1 << i));
+
+                const bool doesMemoryHaveDesirableProperites =
+                    (vk_physicalDeviceMemoryProperties.memoryTypes[i].propertyFlags
+                        & desiredMemoryProperties) == desiredMemoryProperties;
+                if (isMemoryTypeSupportedForResource && doesMemoryHaveDesirableProperites)
+                {
+                    memoryTypeIndex = i;
+                    break;
+                }
+            }
+        }
+        assert(memoryTypeIndex != ~0u);
+#endif
+
+        VkMemoryAllocateInfo vk_allocateInfo = { VK_STRUCTURE_TYPE_MEMORY_ALLOCATE_INFO };
+        vk_allocateInfo.pNext = nullptr; // No extensions for now
+        vk_allocateInfo.allocationSize = additionalReqs.vulkanReqs.size;
+        vk_allocateInfo.memoryTypeIndex = 2u; // Todo(achal): LOL?!
+
+        VkDeviceMemory vk_deviceMemory;
+        if (vkAllocateMemory(m_vkdev, &vk_allocateInfo, nullptr, &vk_deviceMemory) == VK_SUCCESS)
+        {
+            return core::make_smart_refctd_ptr<CVulkanMemoryAllocation>(this, vk_deviceMemory);
+        }
+        else
+        {
+            return nullptr;
+        }
+    }
+
+    //! If cannot or don't want to use device local memory, then this memory can be used
+    /** If the above fails (only possible on vulkan) or we have perfomance hitches due to video memory oversubscription.*/
+    core::smart_refctd_ptr<IDriverMemoryAllocation> allocateSpilloverMemory(
+        const IDriverMemoryBacked::SDriverMemoryRequirements& additionalReqs) override
+    {
+        return nullptr;
+    }
+
+    //! Best for staging uploads to the GPU, such as resource streaming, and data to update the above memory with
+    core::smart_refctd_ptr<IDriverMemoryAllocation> allocateUpStreamingMemory(
+        const IDriverMemoryBacked::SDriverMemoryRequirements& additionalReqs) override
+    {
+        return nullptr;
+    }
+
+    //! Best for staging downloads from the GPU, such as query results, Z-Buffer, video frames for recording, etc.
+    core::smart_refctd_ptr<IDriverMemoryAllocation> allocateDownStreamingMemory(
+        const IDriverMemoryBacked::SDriverMemoryRequirements& additionalReqs) override
+    {
+        return nullptr;
+    }
+
+    //! Should be just as fast to play around with on the CPU as regular malloc'ed memory, but slowest to access with GPU
+    core::smart_refctd_ptr<IDriverMemoryAllocation> allocateCPUSideGPUVisibleMemory(
+        const IDriverMemoryBacked::SDriverMemoryRequirements& additionalReqs) override
+    {
+        return nullptr;
     }
```

Comment:
for this you get `ILogicalDevice::get**MemoryReqs()` and patch up the values that dont make sense (mostly `vulkanReqs`) with `additionalReqs`:
- overwrite size
- overwrite alignment
- bitwise AND the memory type bits
- overwrite the dedicated allocation stuff

Reply:
resolved

Reply:
invalidated by new design!